server/cherry_pick.go

```diff
@@ -23,6 +23,51 @@ import (
 	"github.com/pkg/errors"
 )
 
+const (
+	cherryPickScheduledMsg = "Cherry pick is scheduled."
+	tooManyCherryPickMsg   = "There are too many cherry picking requests. Please do this manually or try again later."
+)
+
+type cherryPickRequest struct {
+	pr        *model.PullRequest
+	version   string
+	milestone int
+}
+
+func (s *Server) listenCherryPickRequests() {
+	defer func() {
+		close(s.cherryPickStoppedChan)
+	}()
+
+	for job := range s.cherryPickRequests {
+		func() {
+			ctx, cancel := context.WithTimeout(context.Background(), defaultRequestTimeout*time.Second)
+			defer cancel()
+			pr := job.pr
+			cmdOut, err := s.doCherryPick(ctx, strings.TrimSpace(job.version), &job.milestone, pr)
+			if err != nil {
+				msg := fmt.Sprintf("Error trying doing the automated Cherry picking. Please do this manually\n\n```\n%s\n```\n", cmdOut)
+				s.sendGitHubComment(ctx, pr.RepoOwner, pr.RepoName, pr.Number, msg)
+				mlog.Error("Error while cherry picking", mlog.Err(err))
+			}
+		}()
+	}
+}
+
+func (s *Server) finishCherryPickRequests() {
+	close(s.cherryPickRequests)
+	select {
+	case <-time.After(defaultRequestTimeout * time.Second):
+		ctx, cancel := context.WithTimeout(context.Background(), defaultRequestTimeout*time.Second)
+		defer cancel()
+		for job := range s.cherryPickRequests {
```

Comment:
Hmm .. this is okay, but it races with the original goroutine. So what we are doing is essentially speeding up the shutdown. That's okay I guess. But let's add a comment to explain what is going on here.